enumerable_methods.rb

```diff
@@ -0,0 +1,237 @@
+        value = self[key]
+        key_value = [key, value]
+        yield(key_value, item)
+      end
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_arr = []
+    i = 0
+    while i < size
+      new_arr << self[i] if yield(self[i])
+      i += 1
+    end
+    new_arr
+  end
+
+  # my_all
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_all?(pattern = nil)
+    if !block_given? && pattern.nil?
+      length.times do |i|
+        return false unless i
+      end
+    elsif self.class == Array
+      length.times do |i|
+        return false unless yield(self[i])
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      length.times do |i|
+        return false unless yield(keys[i], self[keys[i]])
+      end
+    elsif pattern.is_a? Regexp
+      length.times do |i|
+        return false unless i =~ pattern
+      end
+    elsif pattern.is_a? Class
+      length.times do |i|
+        return false unless i.is_a? pattern
+      end
+    elsif pattern
+      length.times do |i|
+        return false unless i == pattern
+      end
+    end
+    true
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_any
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_any?(pattern = nil)
+    if !block_given? && pattern.nil?
+      length.times do |i|
+        return true if i
+      end
+    elsif self.class == Array
+      length.times do |i|
+        return true if yield(self[i])
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      keys.length.times do |i|
+        return true if yield(keys[i], self[keys[i]])
+      end
+    elsif pattern.is_a? Regexp
+      length.times do |i|
+        return true if i =~ pattern
+      end
+    elsif pattern.is_a? Class
+      length.times do |i|
+        return true if i.is_a? pattern
+      end
+    elsif pattern
+      length.times do |i|
+        return true if i == pattern
+      end
+    end
+    false
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_none
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_none?(pattern = nil)
+    if !block_given? && pattern.nil?
+      length.times do |i|
+        return false if i
+      end
+    elsif self.class == Array
+      length.times do |i|
+        return false if yield(self[i])
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      keys.length.times do |i|
+        return false if yield(keys[i], self[keys[i]])
+      end
+    elsif pattern.is_a? Regexp
+      length.times do |i|
+        return false if i =~ pattern
+      end
+    elsif pattern.is_a? Class
+      length.times do |i|
+        return false if i.is_a? pattern
+      end
+    elsif pattern
+      length.times do |i|
+        return false if i == pattern
+      end
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true only if none of the collection members is true.
```
Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got false
```
* When a class is passed as an argument returns true if none of the collection is a member of such class.
```
Failure/Error: expect(array.my_none?(String)).to be true
      
      LocalJumpError:
        no block given (yield)
```
* When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      LocalJumpError:
        no block given (yield)
```
* When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      LocalJumpError:
        no block given (yield)
```